### PR TITLE
Added global logging. 

### PR DESCRIPTION
**Pull request by senhorinfinito**
Replace print statements with structured logging using LOGGER.info() for better log management and consistency
Update logging calls to use f-strings and maintain existing log message semantics

Sample of logs Info and warning . 
image

some of the failed  github action 
Screenshot from 2024-12-01 18-01-41
Screenshot from 2024-12-01 18-01-35

Screenshot from 2024-12-01 17-58-30

### CODE REVIEW

File: trolo/inference/detection.py
Comment (AyushExel) [inline trolo/inference/detection.py:46]: make this warning

File: trolo/train.py
Comment (AyushExel) [inline trolo/train.py:112]: can remove this log, it was meant for debugging

**Review by AyushExel — CHANGES_REQUESTED** [review]
Thanks, left a couple comments

File: trolo/train.py
Comment (senhorinfinito) [reply trolo/train.py]: Done

File: trolo/inference/detection.py
Comment (senhorinfinito) [reply trolo/inference/detection.py]: Done

**Comment by senhorinfinito** [comment]
@AyushExel 
applied suggested changes

**Comment by AyushExel** [comment]
Thanks will merge this after merging the gh-test PR

**Comment by senhorinfinito** [comment]
Thanks

**Comment by AyushExel** [comment]
tests failing please fix them

**Comment by senhorinfinito** [comment]
On it

**Comment by senhorinfinito** [comment]
@AyushExel 
stuck with circular update issue, give me some time. Will fix.

**Comment by AyushExel** [comment]
Sure sure take your time.

**Comment by senhorinfinito** [comment]
@AyushExel  Checks are passed, have to move the init due to circular import issue.  Fixed now/  
let me know anything need to be changed .

**Review by AyushExel — APPROVED** [review]

**Pull request merged**
